numpy/f2py/crackfortran.py

```diff
@@ -2605,7 +2605,7 @@ def analyzevars(block):
     params = get_parameters(vars, get_useparameters(block))
 
     dep_matches = {}
-    name_match = re.compile(r'\w[\w\d_$]*').match
+    name_match = re.compile(r'\w[\w$]*').match
```

Comment:
I suspect the original here was wrong.

> A Fortran identifier must satisfy the following rules:
The first character must be a letter, The remaining characters, if any, may be letters, digits, or underscores, Fortran identifiers are case insensitive. That is, Smith, smith, sMiTh, SMiTH, smitH are all identical identifiers.

The first `\w` should probably be `[A-Za-z]`.  The addition of `\d_` is consistent with misunderstanding of `\w` not to include digits and underscores. The `$` could probably be omitted, but it seems to work as is.

Reply:
Thanks, I revised based on your feedback